core/common/test/format/DateTimeComponentsFormatTest.kt

```diff
@@ -198,6 +199,56 @@ class DateTimeComponentsFormatTest {
         assertNull(bagWithAlternative.nanosecond)
     }
 
+    @Test
+    fun testFormattingWithUnsetFields() {
+        class PropertyAndItsValue<Target, Value>(val property: KMutableProperty1<Target, Value>, val value: Value) {
+            fun set(target: Target) {
+                property.set(target, value)
+            }
+        }
+        val fields = listOf<PropertyAndItsValue<DateTimeComponents, *>>(
+            PropertyAndItsValue(DateTimeComponents::timeZoneId, "Europe/Berlin"),
+            PropertyAndItsValue(DateTimeComponents::year, 2020),
+            PropertyAndItsValue(DateTimeComponents::monthNumber, 3),
+            PropertyAndItsValue(DateTimeComponents::dayOfMonth, 16),
+            PropertyAndItsValue(DateTimeComponents::dayOfWeek, DayOfWeek.MONDAY),
+            PropertyAndItsValue(DateTimeComponents::dayOfYear, 76),
+            PropertyAndItsValue(DateTimeComponents::hour, 23),
+            PropertyAndItsValue(DateTimeComponents::hourOfAmPm, 11),
+            PropertyAndItsValue(DateTimeComponents::amPm, AmPmMarker.PM),
+            PropertyAndItsValue(DateTimeComponents::minute, 59),
+            PropertyAndItsValue(DateTimeComponents::second, 45),
+            PropertyAndItsValue(DateTimeComponents::nanosecond, 123_456_789),
+            PropertyAndItsValue(DateTimeComponents::offsetHours, 3),
+            PropertyAndItsValue(DateTimeComponents::offsetMinutesOfHour, 30),
+            PropertyAndItsValue(DateTimeComponents::offsetSecondsOfMinute, 15),
```

Comment:
Is `offsetIsNegative` false by default?
Or null in `offsetIsNegative` is equivalent to false when formatting?

Reply:
> Or null in `offsetIsNegative` is equivalent to false when formatting?

This one. We haven't explicitly designed this behavior, so we can discuss it if you think it's problematic.